src/materialized/src/bin/materialized/main.rs

```diff
@@ -127,6 +127,22 @@ struct Args {
     #[structopt(long)]
     disable_persistent_system_tables_test: bool,
 
+    /// An S3 bucket used to persist data. If unset, files stored under
+    /// `--data-directory/-D` are used instead. Ignored if persistence is
+    /// disabled.
+    ///
+    /// If set, S3 credentials and region must be available in the process or
```

Comment:
@benesch is there a good stock explanation for this somewhere that I could point to/reuse?

Also, do you have any opinions on how users should configure storage? It seems like these flags will get unwieldy. I was hoping to be able to draw inspiration from how S3 sources do it, but it wasn't clear to me how the precedent would apply

Reply:
> Yeah, https://github.com/rusoto/rusoto/blob/master/AWS-CREDENTIALS.md.

Done!

> Opened a conversation in Slack about the questions about configuration!

Took inspiration from your snowflake suggestion for the combined bucket+path syntax